.github/workflows/test.yml

```diff
@@ -0,0 +1,42 @@
+name: Run all tests & checks
+
+on:
+  push:
+    branches: [ main, nightly ]
+  pull_request:
+    branches: [ main, nightly ]
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+
+      - name: Upgrade pip
+        run: python -m pip install --upgrade pip setuptools==44.0.0
```

Comment:
Same question as the other PR about why we're pinning here instead of using pip.txt like the other actions

Reply:
Addressed!